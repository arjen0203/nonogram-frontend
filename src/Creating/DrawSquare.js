import React, {Component} from 'react';

class DrawSquare extends Component {
    render() {
        let style;
        if (this.props.value === 1){
            style = "draw-square-filled"
        } else {
            style = "draw-square";
        }
        return (
            <div data-testid="drawSquare"
                 className={style}
                 onMouseDown={(e) => this.props.onMouseDown(e)}
                 onMouseEnter={(e) => this.props.onMouseEnter(e)}
            >
            </div>
        );
    }
}

export default DrawSquare;