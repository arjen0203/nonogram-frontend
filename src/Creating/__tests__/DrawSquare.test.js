import React from "react";
import ReactDOM from 'react-dom';
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"

import renderer from "react-test-renderer";
import DrawSquare from "../DrawSquare";

afterEach(cleanup);

it('Renders without crashing', function () {
	const div = document.createElement('div');
	ReactDOM.render(<DrawSquare></DrawSquare>, div);
});

it('Render correctly empty', function () {
	const val = 0;

	const comp = render(<DrawSquare value={val}></DrawSquare>).getByTestId("drawSquare");
	expect(comp).toBeEmpty();
	expect(comp).toHaveClass("draw-square");
});

it('Render correctly filled in', function () {
	const val = 1;

	const comp = render(<DrawSquare value={val}></DrawSquare>).getByTestId("drawSquare");
	expect(comp).toBeEmpty();
	expect(comp).toHaveClass("draw-square-filled");
});

it('matches snapshot', function () {
	const tree = renderer.create(<DrawSquare></DrawSquare>).toJSON();
	expect(tree).toMatchSnapshot();
});