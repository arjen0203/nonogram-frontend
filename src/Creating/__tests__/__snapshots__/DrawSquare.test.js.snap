// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="draw-square"
  data-testid="drawSquare"
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
/>
`;
